packages/ra-ui-materialui/src/input/AutocompleteInput/index.js

```diff
@@ -0,0 +1,396 @@
+ * By default, the options are built from:
+ *  - the 'id' property as the option value,
+ *  - the 'name' property an the option text
+ * @example
+ * const choices = [
+ *    { id: 'M', name: 'Male' },
+ *    { id: 'F', name: 'Female' },
+ * ];
+ * <AutocompleteInput source="gender" choices={choices} />
+ *
+ * You can also customize the properties to use for the option name and value,
+ * thanks to the 'optionText' and 'optionValue' attributes.
+ * @example
+ * const choices = [
+ *    { _id: 123, full_name: 'Leo Tolstoi', sex: 'M' },
+ *    { _id: 456, full_name: 'Jane Austen', sex: 'F' },
+ * ];
+ * <AutocompleteInput source="author_id" choices={choices} optionText="full_name" optionValue="_id" />
+ *
+ * `optionText` also accepts a function, so you can shape the option text at will:
+ * @example
+ * const choices = [
+ *    { id: 123, first_name: 'Leo', last_name: 'Tolstoi' },
+ *    { id: 456, first_name: 'Jane', last_name: 'Austen' },
+ * ];
+ * const optionRenderer = choice => `${choice.first_name} ${choice.last_name}`;
+ * <AutocompleteInput source="author_id" choices={choices} optionText={optionRenderer} />
+ *
+ * The choices are translated by default, so you can use translation identifiers as choices:
+ * @example
+ * const choices = [
+ *    { id: 'M', name: 'myroot.gender.male' },
+ *    { id: 'F', name: 'myroot.gender.female' },
+ * ];
+ *
+ * However, in some cases (e.g. inside a `<ReferenceInput>`), you may not want
+ * the choice to be translated. In that case, set the `translateChoice` prop to false.
+ * @example
+ * <AutocompleteInput source="gender" choices={choices} translateChoice={false}/>
+ *
+ * The object passed as `options` props is passed to the material-ui <AutoComplete> component
+ *
+ * @example
+ * <AutocompleteInput source="author_id" options={{ fullWidth: true }} />
+ */
+const AutocompleteInput = ({
+    classes: classesOverride,
+    allowEmpty,
+    choices,
+    fullWidth,
+    helperText,
+    label,
+    limitChoicesToValue,
+    onBlur,
+    onChange,
+    onFocus,
+    options,
+    optionText,
+    optionValue,
+    resource,
+    setFilter,
+    shouldRenderSuggestions: shouldRenderSuggestionsOverride,
+    source,
+    suggestionComponent,
+    translateChoice,
+    validate,
+    ...rest
+}) => {
+    const translate = useTranslate();
+    const classes = useStyles({ classes: classesOverride });
+    const {
+        id,
+        input,
+        isRequired,
+        meta: { touched, error },
+    } = useInput({
+        onBlur,
+        onChange,
+        onFocus,
+        resource,
+        source,
+        type: 'checkbox',
+        validate,
+        ...rest,
+    });
+
+    let inputEl = useRef();
+    let anchorEl = useRef();
+
+    const updateFilter = useCallback(
+        value => {
+            if (setFilter) {
+                setFilter(value);
+            }
+        },
+        [setFilter]
+    );
+
+    useEffect(() => {
+        updateFilter('');
+    }, [input.value, updateFilter]);
```

Comment:
I don't understand this one. Each time the value changes, the filter resets? Why? A JsDoc comment would be welcome

Reply:
I added a comment about it